storage/innobase/xtrabackup/src/keyring_plugins.cc

```diff
@@ -508,17 +519,19 @@ bool xb_tablespace_keys_load_one(const char *dir, const char *transition_key,
     msg_ts("Error reading %s: failed to read salt.\n", XTRABACKUP_KEYS_FILE);
     goto error;
   }
-
-  if (fread(transition_key_name, 1, sizeof(transition_key_name), f) !=
-      sizeof(transition_key_name)) {
+  transition_key_name =
+      (char *)my_malloc(PSI_NOT_INSTRUMENTED, transition_key_name_size,
+                        MYF(MY_FAE | MY_ZEROFILL));
```

Comment:
possible to used std::unique_ptr?

Reply:
or the variant with custom deleter (we can use my_free() there)

Reply:
@satya-bodapati good suggestion. I've added smart pointer. 

Regarding the 2.4, I will do the changes there too. However, since the 5.7 and 8.0 commits related to vault v2 are different, I've not merged the 2.4 branch into the 8.0. I had to cherry pick them individually. I will do a null merge after both branches are merged.